### ISSUE TITLE
Conflicts with NSLogger

### ISSUE THREAD

**Issue by jeremiegirault** [Instabug/Instabug-iOS#158]
I want to report that the use of Instabug with https://github.com/fpillet/NSLogger causes an app kill.
Usage of either one of the libs is fine, the crash only happens when using both. 

This may be related to a loop where NSLogger sends logs over the network and Instabug logs the network messages. I could not find how to disable network inspection in Instabug.

**Steps to reproduce:**
Use NSLogger and Instabug as Pod. Log a message using NSLogger
**Expected behavior:**
Message is logged, app is running.
**Actual Behavior:**
App is terminated with signal 5.

cc @fpillet

**Comment by fpillet** [comment]
I'm the author of NSLogger. It intercepts logs, queues them to a pool and sends them over the network to a desktop viewer. I'm not sure if there is special machinery in Instabug to isolate specific connections -- if there is, I can mark NSLogger's connection so as to avoid the problem above.

**Comment by HeshamMegid** [comment]
As @fpillet pointed out, this happens because Instabug intercepts network requests made by NSLogger, which results in a endless loop.

Please check [our docs](http://docs.instabug.com/docs/network-requests-logging#section-omitting-requests-from-logs) on how to filter out certain requests from being logged. We're also looking into this to see if there's anything we could do from our side.

**Comment by fpillet** [comment]
@HeshamMegid what's interesting is that NSLogger uses direct TCP connections (no HTTP). It looks over Bonjour to find the viewer (unless you give it a direct IP address) then connects to the one if finds. I don't think your filtering API would work here.

Note that logs are always pushed to the network from a single private thread created by NSLogger. I could set a thread dictionary key on this thread to indicate that it's NSLogger and that you should avoid intercepting its connections. Let me know if it's a suitable solution.

**Comment by HeshamMegid** [comment]
@fpillet that sounds like something that shouldn't go through our network logger in the first place. Let me look into this more and get back to you.

**Comment by jeremiegirault** [comment]
@HeshamMegid I tried to use: `Instabug.setNetworkLoggingEnabled(false)` before initializing instable as I don't have a use for this feature but it's still conflicting in some ways.

**Comment by Kmohamed** [comment]
@jeremiegirault Did you see this message before the app being killed ?
![screen shot 2017-05-19 at 2 24 57 am](https://cloud.githubusercontent.com/assets/1115748/26228489/91cdd840-3c3a-11e7-9e3a-026bbc3ead2c.png)

**Comment by fpillet** [comment]
@Kmohamed we did some shared screen debugging so I witnessed the session and no, the message didn't appear. The OS simply jettisons the app on locking, with a signal 5.

**Comment by jeremiegirault** [comment]
I force the bug again and yes, constated this popup when the app is killed

**Comment by Kmohamed** [comment]
@fpillet I could not find where is the loop also I could not find see the stacktrace it's very weird 
@jeremiegirault The app is killed If it's attached to Xcode only am I right ?

**Comment by jeremiegirault** [comment]
Indeed, the freeze/kill only happens when the debugger is attached. Strange indeed

**Comment by Figo0o** [comment]
Well, After digging deeper into this I've found where is the problem is. The Instabug Crash Reporting got enabled although the debugger is attached which causes the Freezing. 

Somehow our checks for the debugger attached fails because of NSLogger. Does NSLogger do changes to the stderr's files because we can't find them after Integrating. @fpillet any idea ?

@jeremiegirault for now you can disable Instabug Crash Reporting. This should solve your problem in the current time  (only if the debugger is attached) until we found a solution for that.

**Comment by jeremiegirault** [comment]
@Figo0o I tried something like this without success: 
```
#if DEBUG
            Instabug.setCrashReportingEnabled(false)
#endif
Instabug.start(withToken: Keys.shared.instabug, invocationEvent: .shake)
```

For your info I think NSLogger catches stderr and stdout to show these logs in the app. This link may help you: https://github.com/fpillet/NSLogger/blob/master/Client/iOS/LoggerClient.m#L1226

**Comment by Figo0o** [comment]
@jeremiegirault You have to disable crash reporting after calling `Instabug.start(withToken: Keys.shared.instabug, invocationEvent: .shake)`. Give it a try and let me know. 

Let me check that.

**Comment by jeremiegirault** [comment]
Workaround ok on my side. Let me know what you find about these std streams.

**Comment by fpillet** [comment]
@Figo0o yes NSLogger intercepts `stderr` and `stdout` (using `dup2`), captures their output and logs everything that comes out of them.

**Comment by fpillet** [comment]
@Figo0o relevant code (and interception function) are on line https://github.com/fpillet/NSLogger/blob/master/Client/iOS/LoggerClient.m#L1284 and around

**Comment by Figo0o** [comment]
@fpillet I'm trying to figure this out. `dup` deletes the source file. maybe thats the problem. am I missing something?

**Comment by Figo0o** [comment]
@jeremiegirault 

We have just released Instabug [V7.2.8](https://github.com/Instabug/Instabug-iOS/releases/tag/v7.2.8) with the fix for this issue. Please give it a try and let me know your feedback. 

Thanks!
